### ISSUE TITLE
Should we be able to use the `validate` parser to do validation?

### ISSUE THREAD

**Issue by kbarrette** [ohler55/oj#788]
I have a use case where I'd like a fast validation of JSON, but it seems like the `validate` parser accepts bad JSON?

```
ruby -rOj -e 'Oj::Parser.validate.parse("{")'
[no output]

ruby -rOj -e 'Oj::Parser.usual.parse("{")'
-e:1:in `<main>': Stack consistency error (sp: 5, bp: 6) (fatal)
== disasm: #<ISeq:<main>@-e:1 (1,0)-(1,27)> (catch: FALSE)
0000 opt_getinlinecache                     13, <is:0>                (   1)[Li]
0003 putobject                              true
0005 getconstant                            :Oj
0007 putobject                              false
0009 getconstant                            :Parser
0011 opt_setinlinecache                     <is:0>
0013 opt_send_without_block                 <calldata!mid:usual, argc:0, ARGS_SIMPLE>
0015 putstring                              "{"
0017 opt_send_without_block                 <calldata!mid:parse, argc:1, ARGS_SIMPLE>
0019 leave
```

**Comment by ohler55** [comment]
Please take a look at the detect-not-closed branch. If it works for you I'll release.

**Comment by ohler55** [comment]
Released v3.13.17 with the fix. Please check and close this issue if it works for you.

**Comment by ohler55** [comment]
No response and unit test indicate the issue is fixed, closing.
